### PR TITLE
⬆️ deps: update argminmax to 0.6.1

### LINKED ISSUE #50: :arrow_up: update to latest argminmax release
Currently this library pins the argminmax dependency to `v0.3.1` - which uses [`ndarray::ArrayView1`](https://docs.rs/ndarray/latest/ndarray/type.ArrayView1.html) as underlying and mandatory interface.

https://github.com/predict-idlab/tsdownsample/blob/33a7cfacdae924b813ea1ee13914160092a5ae84/downsample_rs/Cargo.toml#L12

In more recent releases of argminmax I switched to using slices internally and offer *optional* compatibility with other data types, including `ndarray::ArrayView1`.

Updating the argminmax dependency to the latest version would thus allow to remove the `ndarray` dependency (as the rust-ndarray crate is not so well maintained + a quite expensive build dependency). 
Moreover, the latest versions of argminmax contain some further optimizations and new features (e.g., nan handling -> updating the argminmax dependency is thus necessary to tackle https://github.com/predict-idlab/tsdownsample/issues/24)

### PR DESCRIPTION
**Pull request by NielsPraet**
Closes #50

### CODE REVIEW

**Comment by jonasvdd** [comment]
Ready for review? @NielsPraet @jvdd?

**Comment by NielsPraet** [comment]
@jonasvdd now it should be!

File: downsample_rs/src/lttb/scalar.rs [hunk 1]
```diff
@@ -16,16 +16,16 @@ fn f64_to_i64unsigned(v: f64) -> i64 {
...
-    x: ArrayView1<Tx>,
-    y: ArrayView1<Ty>,
+    x: &[Tx],
+    y: &[Ty],
     n_out: usize,
-) -> Array1<usize>
+) -> Vec<usize>
 where
-    for<'a> ArrayView1<'a, Ty>: Average,
+    for<'a> &'a [Ty]: ArgMinMax,
```
Comment (jvdd) [inline downsample_rs/src/lttb/scalar.rs:24]: Why is this necessary here?

File: downsample_rs/src/lttb/scalar.rs [hunk 2]
```diff
@@ -103,12 +101,12 @@ where
 
 // ----------- WITHOUT X
 
-pub fn lttb_without_x<Ty: Num + AsPrimitive<f64>>(y: ArrayView1<Ty>, n_out: usize) -> Array1<usize>
+pub fn lttb_without_x<Ty: Num + AsPrimitive<f64>>(y: &[Ty], n_out: usize) -> Vec<usize>
 where
-    for<'a> ArrayView1<'a, Ty>: Average,
+    for<'a> &'a [Ty]: ArgMinMax,
```
Comment (jvdd) [inline downsample_rs/src/lttb/scalar.rs:106]: Same as above

File: downsample_rs/src/lttb/scalar.rs (hunk 1 above)
Comment (NielsPraet) [reply downsample_rs/src/lttb/scalar.rs:24]: It's not... small oversight!
